### ISSUE TITLE
React-Native support

### ISSUE THREAD

**Issue by bkniffler** [segmentio/ui-box#29]
Hey, I love the concept of Evergreen being built on one single primitive. This could also make react-native compatibility a bit easier. 

I'd love to hear if you've ever considered making ui-box react-native compatible.

**Comment by Rowno** [comment]
I have no idea how react-native works so I have no idea how to make ui-box compatible 😅 

I'm assuming react-native doesn't support all of these CSS properties though? https://github.com/segmentio/ui-box#css-properties

**Comment by bkniffler** [comment]
Not all, but in fact a whole lot of them: https://github.com/vhpoet/react-native-styling-cheat-sheet

**Comment by Rowno** [comment]
I guess it would be possible to create a subset of ui-box that has the CSS injection code removed and supports less CSS properties. But if all the react-native primitives already have all these CSS properties, I don't see what benefit ui-box would add?

**Comment by bkniffler** [comment]
I thought that since ui-box is already the big building block of segments ui framework, it would be possible to use some of the components on react-native + web. The thing is, most ui frameworks will never be possible to work on react-native since there is excessive css selectors, even if styled-components is used (see atlas kit for example). Segment already uses the styles in a way that fits 100% to how you'd style react-native components plus 80% of the styles are compatible already (I guess).

**Comment by auderer** [comment]
@Rowno yes! [styled-components](https://github.com/styled-components/styled-components) does this. For web, you import from `styled-components`, and for native you import from `styled-components/native`. It's still the same NPM package.

**Comment by aleclarson** [comment]
FWIW, I'm using [react-primal](https://github.com/alloc/react-primal) (same concept as `ui-box`) in React Native
